src/plugins/intel_npu/src/al/include/sync_infer_request.hpp

```diff
@@ -92,56 +92,32 @@ class SyncInferRequest : public ov::IInferRequest {
      */
     void initialize_states();
 
+protected:
     /**
-     * @return The state tensors accessible by their names.
-     */
-    std::unordered_map<std::string, std::shared_ptr<VariableState>>& get_variable_states() {
-        return _variableStates;
-    }
-
-    /**
-     * @return The names used by the inputs in the order registered inside the model.
-     */
-    std::vector<std::string> get_input_names() {
-        return _metadata.inputNames;
-    }
-
-    /**
-     * @return The names used by the outputs in the order registered inside the model.
-     */
-    std::vector<std::string> get_output_names() {
-        return _metadata.outputNames;
-    }
-
-    /**
-     * @return The names used by the state variables in the order registered inside the model.
-     */
-    std::vector<std::string> get_state_names() {
-        return _metadata.stateNames;
-    }
-
-    /**
-     * @return The names used by the shape variables in the order registered inside the model.
+     * @see ov::ISyncInferRequest
      */
-    std::vector<std::string> get_shape_names() {
-        return _metadata.shapeNames;
-    }
+    struct FoundPort {
+        size_t idx;
+        enum class Type { NOT_FOUND = 0, INPUT, OUTPUT } type;
 
-    /**
-     * @return A map holding references towards all tensors used by the current inference request object.
-     */
-    std::unordered_map<std::string, std::shared_ptr<ov::ITensor>>& get_all_tensors() {
-        return _allTensors;
-    }
+        bool found() {
+            return type != Type::NOT_FOUND;
+        }
+        bool is_input() {
+            return type == Type::INPUT;
+        }
+        bool is_output() {
+            return !is_input();
+        }
+    };
 
     /**
-     * @return A map holding references towards all shapes tensors used by the current inference request object.
+     * @brief Finds input or output port
+     * @return structure which contains index of Input/Output or report that port wasn't found
+     * @see ov::ISyncInferRequest
      */
-    std::unordered_map<std::string, std::shared_ptr<ov::ITensor>>& get_shapes_tensors() {
-        return _shapesTensors;
-    }
+    FoundPort find_port(const ov::Output<const ov::Node>& port) const;
```

Comment:
Do we expect this to be used in IMD? If not let's consider moving into ZeroInferRequetst

Reply:
Yeah, it is essential for IMD too. The method is used inside the `get_tensor` and `set_tensor` functions which are the standard means of interacting with the inputs/outputs used by the backend.